image/store_test.go

```diff
@@ -70,6 +72,7 @@ func TestRestore(t *testing.T) {
 
 	invalidPattern := id1.Encoded()[1:6]
 	_, err = imgStore.Search(invalidPattern)
+	assert.Check(t, errdefs.IsNotFound(err), "unexpected error type %T", err)
 	assert.ErrorContains(t, err, "No such image")
```

Comment:
As we're now checking the error-type to be a "not found", I wonder if we still need to check the `No  such image`. If we _do_ want to keep it (to make sure it's a 404 related to the _image_ and not anything else), perhaps more relevant would be if the error contains the expected image-id;

```suggestion
	assert.Check(t, is.ErrorContains(err, invalidPattern))
```

Short background why I was looking;

Before we had ("proper") typed errors, there was a horrible hack in the code that did string-matching to convert errors to HTTP status codes. So at the time, a "not found" error MUST contain `No such`, so that it's converted to a `404`. We got rid of that, but still left that convention for some time (as there were (sigh) other project that used the same string matching 😬)

Basically; I _think_ I have a branch somewhere that changes them to lowercase, and/or some rephrasing, having the (capital) `No such image` already gone from the test can help for some of these.

Reply:
Good catch, thanks.
Initially I wanted to get rid of the local `imageNotFoundError` helper and inline it to a single `errdefs.NotFound` call, but since the message starts with uppercase the linter was complaining about it, and I assumed that we need to keep it this way for backwards compatibility 😄